ngraph/frontend/generic/src/plugin_loader.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#ifdef _WIN32
+#include <Windows.h>
+#include <direct.h>
+#else // _WIN32
+#include <dirent.h>
+#include <dlfcn.h>
+#include <unistd.h>
+#endif // _WIN32
+
+#include <string>
+#include <sys/stat.h>
+#include <vector>
+#include "ngraph/file_util.hpp"
+
+#include "plugin_loader.hpp"
+
+using namespace ngraph;
+using namespace ngraph::frontend;
+
+#ifdef WIN32
+#define DLOPEN(fileStr) LoadLibrary(TEXT(fileStr.c_str()))
```

Comment:
Can we reuse `SharedObjectLoader` from the Inference Engine?

Reply:
@ilyachur @ilya-lavrenov 
This question is the same as https://github.com/openvinotoolkit/openvino/pull/5470#discussion_r623645020
My opinion is that it is possible to reuse only ~3 lines of macro, and it doesn't look worth to have additional common component for ngraph & inference-engine.
----
Approach to have separate library with SharedObjectLoader (located somewhere in common location for IE and Ngraph) looks too complicated comparing to such #ifdef wrappers around dlopen/dlclose/dlsym
Rest of functionality is frontend-specific, e.g. look for a directory, find plugins, call methods, cast results, return handles, etc.
----
But probably we can discuss it, if needed